tmt/base.py

```diff
@@ -3034,9 +3054,27 @@ def _convert_node(node: fmf.Tree) -> 'Plan':
 
             self.debug(f"Turning node '{node.name}' into a plan.", level=3)
 
+            fmf_context = FmfContext.from_spec(
+                node.name, node.data.get('context', {}), self._logger
+            )
+
+            if reference.inherit_context:
+                fmf_context.update(self._fmf_context)
+            else:
+                fmf_context.update(self._cli_fmf_context)
```

Comment:
Shouldn't the cli context be applied always? Not only when `inherit_context` is `False`?